### ISSUE TITLE
Equivalent to `git config --list`

### ISSUE THREAD

**Issue by AtkinsSJ** [isomorphic-git/isomorphic-git#1917]
As far as I can tell, there's no ability to list all specified config values - you can only request specific ones by name.

I don't know what would be the best approach for this. Probably returning an array of `{key, value}` would be good enough?

**Comment by jcubic** [comment]
It's not good idea because config file have sections. Probably it would be good idea to check other common libraries for parsing ini files (git config have similar syntax) to see what API they use.

Do you want to contribute? We can think about the API.

**Comment by jcubic** [comment]
Hmm, I would check what canonical git is doing. It probably overwrites the previous value so I would do the same. But I'm not sure what git is doing.

**Comment by jcubic** [comment]
LIPS returns:

```scheme
``(,@,@l ,@,@l)
(quasiquote ((unquote-splicing x y)))
```

It should be:

```scheme
(quasiquote ((unquote-splicing x y) (unquote-splicing x y)))
```

**Comment by AtkinsSJ** [comment]
Canonical git includes all the keys individually. After running these in a repo:
```
git config --local user.email hello
git config --local --add user.email world
```
`.git/config` contains this:
```
[user]
        email = hello
        email = world
```
And `git config --list` includes both of those, as well as the one from the global config, which appears first:
(I know isomorphic-git doesn't support a global config file yet. That's on my list to try and add too.)
```
❯ git config --list
user.email=atkinssj@gmail.com

...snip...

user.email=hello
user.email=world
```

**Comment by jcubic** [comment]
it seems there is [getConfigAll](https://isomorphic-git.org/docs/en/getConfigAll) can you check if it does what you need?

**Comment by AtkinsSJ** [comment]
`getConfigAll()` returns all the values for a given key. So in my example it'd return `[ 'atkinssj@gmail.com', 'hello', 'world' ]`. What I want is to read *every* key and value. There's currently no other way of getting a list of config keys like that.

For context, I'm working on a full CLI git client that runs on Puter.com. You can find the code [here](https://github.com/HeyPuter/puter/tree/main/packages/git). The goal is for it to replicate canonical git as closely as possible.

For my purposes, a `git.listConfig()` (with options for which source to use, once isomorphic-git has sources other than the local config file) that returns an array of `{ key, value }` would be good. I can get started on that if it sounds OK to you.

**Comment by jcubic** [comment]
yes, you can work on this. But don't change the default behavior, we don't want breaking changes, even if the result of that function is not very useful.

**Comment by AtkinsSJ** [comment]
I'm not going to be able to work on this after all, but I'll leave the issue open.
